cmd/limactl/shell.go

```diff
@@ -151,7 +150,20 @@ func shellAction(cmd *cobra.Command, args []string) error {
 		"--",
 		script,
 	}...)
-	sshCmd := exec.Command(arg0, sshArgs...)
+
+	var sshCmd *exec.Cmd
+
+	if usrShell := os.Getenv(envShell); usrShell != "" {
+		sshCmdStr := shellescape.QuoteCommand(append([]string{"ssh"}, sshArgs...))
+		sshCmd = exec.Command(usrShell, "-i", "-l", "-c", sshCmdStr)
```

Comment:
Ideally, instead of always passing `-i` and `-l`, we should detect whether the current shell is actually a _login_ shell, and whether it is _interactive_.
Unfortunately, there doesn't seem to exist a portable way to do that: [How to check if a shell is login/interactive/batch](https://unix.stackexchange.com/q/26676/152409) (StackExchange - Unix & Linux).

Reply:
Note that `csh` (is anybody still using that?) does not support the `-l` option unless it is the only option given.

Reply:
Right, good point. [csh(1)](https://manpages.org/csh) also mentions

> **-i**
> [...] Shells are interactive without this option if their inputs and outputs are terminals.

Which makes `-i` unnecessary. But passing it explicitly doesn't seem to cause any side effect.

If `csh` is the only exception I guess we can have a simple check for `basename == csh`?

Reply:
> If csh is the only exception

`tcsh` works the same as `csh` 😄

Reply:
Dang.
Let's introduce a trivial `isCsh()` helper then? It's hard to imagine covering all edge cases, but at least those ones should be fairly easy to get right.

Reply:
> `IsCsh()`

`isBash() || isZsh()`  might be safer

Reply:
`|| isFish()` 🐠

Reply:
That sounds a bit restrictive to me to assume Lima users are limited to Bash, Zsh and Fish users.

Although I agree that might cover 95% of cases or more, simply checking for `sh|ash|bash|zsh|fish` doesn't cover the full range of options out there, whereas eliminating `/*csh` seems to eliminate the few exceptions that don't support `-il` well.

You decide.

Reply:
Current status: [60b58339..f4ebbc67](https://github.com/lima-vm/lima/compare/60b58339fe79cb88018a37b50b2a45b55560bc35..f4ebbc6795cdd26a35f70a5d22c8932e0aa63941)

Demo with tcsh (macOS), notice `-l` was omitted:

```
DEBU[0000] executing ssh (may take a long)): [/bin/csh -i -c ssh -F /dev/null -o 'Identi...
```

Happy to revisit.

Reply:
Marking as resolved because the original comment was about whether or not to rely on `IsTerminal()`.

Let's open a new thread to discuss further concerns with shell type detection (if any).